app/src/main/java/com/google/samples/apps/sunflower/compose/PlantDetailView.kt

```diff
@@ -0,0 +1,505 @@
+    val plantDetailsViewModel: PlantDetailViewModel = viewModel(
+        factory = InjectorUtils.providePlantDetailViewModelFactory(ContextAmbient.current, plantId)
+    )
+    val plant = plantDetailsViewModel.plant.observeAsState().value
+    val isPlanted = plantDetailsViewModel.isPlanted.observeAsState().value
+
+    // Every time there's a new value for plant or isPlanted LiveData, this block will get executed
+    if (plant != null && isPlanted != null) {
+        val context = ContextAmbient.current
+        val view = ViewAmbient.current
+
+        PlantDetails(plant, isPlanted, PlantDetailsCallbacks(
+            onBackClicked = onBackClicked,
+            onFabClicked = {
+                plantDetailsViewModel.addPlantToGarden()
+                Snackbar.make(view, R.string.added_plant_to_garden, Snackbar.LENGTH_LONG).show()
+            },
+            onShareClicked = {
+                val shareText = context.resources.getString(R.string.share_text_plant, plant.name)
+                onShareClicked(shareText)
+            }
+        ))
+    }
+}
+
+@VisibleForTesting
+@Composable
+fun PlantDetails(
+    plant: Plant,
+    isPlanted: Boolean,
+    callbacks: PlantDetailsCallbacks,
+    modifier: Modifier = Modifier
+) {
+    // PlantDetails owns the scrollerPosition to simulate CollapsingToolbarLayout's behavior
+    val scrollerPosition = ScrollerPosition()
+    var toolbarState by state { ToolbarState.HIDDEN }
+
+    // Transition that fades in/out the header with the image and the Toolbar
+    Transition(
+        definition = toolbarTransitionDefinition,
+        toState = toolbarState
+    ) { transitionState ->
+        Stack(modifier) {
+            PlantDetailsContent(
+                scrollerPosition = scrollerPosition,
+                toolbarShown = toolbarState.toolbarShown,
+                onToolbarShownUpdate = { newValue ->
+                    toolbarState = toolbarStateFromBoolean(newValue)
+                },
+                plant = plant,
+                isPlanted = isPlanted,
+                callbacks = callbacks,
+                transitionState = transitionState
+            )
+            PlantHeader(toolbarState.toolbarShown, plant.name, callbacks, transitionState)
+        }
+    }
+}
+
+@Composable
+private fun PlantDetailsContent(
+    scrollerPosition: ScrollerPosition,
+    toolbarShown: Boolean,
+    onToolbarShownUpdate: (Boolean) -> Unit,
+    plant: Plant,
+    isPlanted: Boolean,
+    callbacks: PlantDetailsCallbacks,
+    transitionState: TransitionState
+) {
+    VerticalScroller(scrollerPosition) {
+        // The header transition happens given the _original_ position of the name on the screen
+        var namePosition by state { Float.MAX_VALUE }
+        // Whenever the name position or the scroller position changes,
+        // check if the toolbar should be shown or not
+        onCommit(namePosition, scrollerPosition.value) {
```

Comment:
I don't think this needs to be in `onCommit` both are observable so will be re-composed on changes.

Reply:
But I want a non-compose function to be executed when those changes. Isn't `onCommit` meant for this?

Reply:
What non-compose function? The callback? That should just work?!